src/DataGrid.tsx

```diff
@@ -496,17 +480,34 @@ function DataGrid<R, SR, K extends Key>(
    * event handlers
    */
   function handleColumnResize(column: CalculatedColumn<R, SR>, width: number | 'max-content') {
-    if (width === 'max-content') {
-      setAutoResizeColumn(column);
-      return;
+    const { style } = gridRef.current!;
```

Comment:
Wondering if we can run into any edge cases with this approach. Does not seem like but may be safer to use react api if there is no noticeable performance improvement with the new approach 
https://beta.reactjs.org/learn/manipulating-the-dom-with-refs#best-practices-for-dom-manipulation-with-refs-best-practices-for-dom-manipulation-with-refs